plugin.audio.podcasts/addon.py

```diff
@@ -0,0 +1,369 @@
+
+        elif res.code == 200:
+            data = res.read()
+            return data.decode("utf-8")
+
+        else:
+            raise HttpStatusError("Unexpected HTTP Status %i for podcast %s" % (res.code, url))
+
+    def _loadRss(self, url):
+
+        _data = self._requestHttp(url)
+        return xmltodict.parse(_data)
+
+    def _load_rss_feed(self, url):
+
+        rss_feed = self._loadRss(url)
+
+        channel = rss_feed["rss"]["channel"]
+
+        if "image" in channel and "url" in channel["image"]:
+            image = channel["image"]["url"]
+        elif "itunes:image" in channel:
+            image = channel["itunes:image"]["@href"]
+        else:
+            image = ""
+
+        items = channel["item"]
+
+        return items, image
+
+    def playRss(self, parent, path, params):
+
+        try:
+            url = params["url"][0]
+            items, image = self._load_rss_feed(url)
+            index = int(params["index"][0])
+
+            item = items[index]
+            if "enclosure" in item:
+                stream_url = item["enclosure"]["@url"]
+            else:
+                stream_url = item["guid"]
+
+            xbmc.executebuiltin('PlayMedia(%s)' % stream_url)
+
+        except HttpStatusError as error:
+
+            xbmc.executebuiltin("Notification(\"HTTP Status Error\", \"%s\")"
+                                % error.message)
+
+    def renderRss(self, parent, path, params):
+
+        url = params["url"][0]
+        try:
+            items, image = self._load_rss_feed(url)
+
+        except HttpStatusError as error:
+
+            xbmc.executebuiltin("Notification(\"HTTP Status Error\", \"%s\")"
+                                % error.message)
+        
+        index = 0
+
+        entries = []
+
+        for item in items:
+
+            #       Does not work: see https://forum.kodi.tv/showthread.php?tid=112916
+            #            if 'pubDate' in item:
+            #                pubDate = datetime.strptime(item['pubDate'][:-6], '%a, %d %b %Y %H:%M:%S')
+            #            else:
+            pubDate = None
+
+            entries = entries + [{
+                "path": str(index),
+                "name": item["title"],
+                "name2": item["description"] if "description" in item else "",
+                "icon": image,
+                "params": [
+                    {
+                        "call": "playRss",
+                        "index": str(index),
+                        "url": url
+                    }
+                ],
+                "pubDate": pubDate
+
+            }]
+
+            index += 1
+
+
+#        entries = sorted(entries, key=itemgetter('pubDate'), reverse=True)
+        for entry in entries:
+
+            self._add_list_item(entry, path)
+
+        xbmcplugin.endOfDirectory(self._addon_handle)
+
+    def play(self, parent, path, params):
+
+        url = params["url"][0]
+        xbmc.executebuiltin('PlayMedia(%s)' % url)
```

Comment:
I usually use `xbmcplugin.setResolvedUrl()` in combination with `xbmcgui.ListItem.setProperty('IsPlayable', 'true')`. That way Kodi will also remember what items were played in the past. But I guess this works. Same in other places.